### ISSUE TITLE
Depthwise Convolutions drive loss to NaN with given model (resnet18)

### ISSUE THREAD

**Issue by dlibenzi** [pytorch/xla#1537]
## 🐛 Bug

Using depthwise convolutions drives the model loss to NaN after 80 steps or so.
To test just change the following value:
```Python
FLAGS['use_dw'] = True
```

Resnet18_DW (XLA):
https://gist.github.com/dlibenzi/c91a72c4817468c43911dce588cfc86f

Resnet18_DW (PT/CPU):
https://gist.github.com/dlibenzi/cc935cc5e3ee3ec12cfdd0ba03235017

On the plus side, recent changes made DW Convs 14x faster! 🎉

**Comment by dlibenzi** [comment]
This happens on TPU as well as CPU device (even fast on CPU actually).

**Comment by zcain117** [comment]
We had a [similar issue](https://github.com/pytorch/xla/issues/870) in the past with losses in conv2d but that was losses decaying down to 0.0, whereas this one is losses exploding to huge numbers right?

**Comment by dlibenzi** [comment]
Yeah.
I feel this might be a "view" (weight sharing) issue.

**Comment by dlibenzi** [comment]
It bombs at step 9 for me:
```
[xla:0](0) Loss=2.48994 Rate=5.35 GlobalRate=5.35 Time=Fri Jan 17 16:18:23 2020
[xla:0](1) Loss=2.35658 Rate=2.45 GlobalRate=0.95 Time=Fri Jan 17 16:20:26 2020
[xla:0](2) Loss=2.56546 Rate=1.30 GlobalRate=0.75 Time=Fri Jan 17 16:22:26 2020
[xla:0](3) Loss=2.71070 Rate=5.75 GlobalRate=0.98 Time=Fri Jan 17 16:22:33 2020
[xla:0](4) Loss=2.50078 Rate=7.73 GlobalRate=1.19 Time=Fri Jan 17 16:22:41 2020
[xla:0](5) Loss=2.52006 Rate=8.36 GlobalRate=1.39 Time=Fri Jan 17 16:22:48 2020
[xla:0](6) Loss=2.74931 Rate=8.73 GlobalRate=1.58 Time=Fri Jan 17 16:22:55 2020
[xla:0](7) Loss=2.46043 Rate=8.85 GlobalRate=1.76 Time=Fri Jan 17 16:23:02 2020
[xla:0](8) Loss=2.54282 Rate=8.86 GlobalRate=1.93 Time=Fri Jan 17 16:23:09 2020
[xla:0](9) Loss=nan Rate=8.97 GlobalRate=2.10 Time=Fri Jan 17 16:23:16 2020
[xla:0](10) Loss=nan Rate=8.98 GlobalRate=2.26 Time=Fri Jan 17 16:23:24 2020
```

**Comment by dlibenzi** [comment]
I played with different things, batch size, disabling momentum, removing sampler, ...
All lead to NAN after the 2nd step.
But, fake data does not lead to NAN:

https://gist.github.com/dlibenzi/45bec23e8124c839a46c5d6aabe7275f

**Comment by dlibenzi** [comment]
This seems to be related to the default `BatchNorm2d` eps being too small.
Setting `eps=1e-2` cured the NANs.
